plugins/modules/postgresql_subscription.py

```diff
@@ -586,15 +586,22 @@ def __get_general_subscr_info(self):
         Returns:
             Dict with subscription information if successful, False otherwise.
         """
+        columns_sub_table = ("SELECT column_name "
+                             "FROM information_schema.columns "
+                             "WHERE table_schema = 'pg_catalog' "
+                             "AND table_name = 'pg_subscription'"
+                             "AND column_name IN ('subenabled','subconninfo','subslotname','subsynccommit','subpublications')")
+        columns_result = exec_sql(self, columns_sub_table, query_params={'name': self.name, 'db': self.db}, add_to_executed=False)
+        columns = ", ".join(["s.%s" % column['column_name'] for column in columns_result])
         query = ("SELECT obj_description(s.oid, 'pg_subscription') AS comment, "
-                 "d.datname, r.rolname, s.subenabled, "
-                 "s.subconninfo, s.subslotname, s.subsynccommit, "
-                 "s.subpublications FROM pg_catalog.pg_subscription s "
+                 "d.datname, r.rolname,"
+                 "%(columns)s "
+                 "FROM pg_catalog.pg_subscription s "
                  "JOIN pg_catalog.pg_database d "
                  "ON s.subdbid = d.oid "
                  "JOIN pg_catalog.pg_roles AS r "
                  "ON s.subowner = r.oid "
-                 "WHERE s.subname = %(name)s AND d.datname = %(db)s")
+                 "WHERE s.subname = '%(name)s' AND d.datname = '%(db)s'" % {'columns': columns, 'name': self.name, 'db': self.db})
```

Comment:
My python is admittedly pretty weak. For some reason doing the string interpolation this way feels wrong to me, but maybe it's fine idk

Reply:
- We are passing `name` and `db` values as paramenters to the execute method a bit below (which is a right way to do it) and we don't need to use the single quotes here AFAIK - it should be handled by the method internally.
- We should somehow substitute only columns here. We could manipulate with string conjugation (if i call it correctly) here to do it.

Reply:
>We should somehow substitute only columns here. 

Agreed. That's what I tried to do initially, but wasn't able to make it work because it was mixing different types of string formatting. I'll play with it a bit more